vaadin-checkbox-flow-parent/vaadin-checkbox-flow/src/test/java/com/vaadin/flow/component/checkbox/tests/CheckboxGroupTest.java

```diff
@@ -328,6 +329,20 @@ public void setItemLabelGenerator_labelIsUpdated() {
         Assert.assertEquals("foo (Updated)", cb.getLabel());
     }
 
+    @Test
+    public void setItemLabelGenerator_removesItemRenderer() {
+        CheckboxGroup<String> checkboxGroup = new CheckboxGroup<>();
+        checkboxGroup.setItems("foo", "bar");
+        checkboxGroup.setRenderer(new TextRenderer<>());
+
+        Assert.assertTrue(
+                checkboxGroup.getItemRenderer() instanceof TextRenderer);
+
+        checkboxGroup.setItemLabelGenerator(item -> item);
+
+        Assert.assertNull(checkboxGroup.getItemRenderer());
```

Comment:
With `RadioButtonGroup`, this wouldn't actually be `null`, but an instance of `TextRenderer`

Reply:
Yeah, that wasn't the goal of this PR, which is mainly about making sure that an item label generator is actually used when setting it after an item renderer.